src/WarcraftLegacies.Source/Quests/Sentinels/QuestFeathermoon.cs

```diff
@@ -34,23 +29,20 @@ public QuestFeathermoon(Capital feathermoon) : base("Shores of Feathermoon",
 
     /// <inheritdoc />
     protected override string RewardDescription => 
-      $"Gain control of all units in Feathermoon Stronghold and learn to train Naisha from the {GetObjectName(UNIT_E00R_ALTAR_OF_WATCHERS_SENTINEL_ALTAR)}";
+      $"Gain the ability to train Naisha from the {GetObjectName(UNIT_E00R_ALTAR_OF_WATCHERS_SENTINEL_ALTAR)}";
 
     /// <inheritdoc />
     protected override void OnFail(Faction completingFaction)
     {
       var rescuer = completingFaction.ScoreStatus == ScoreStatus.Defeated
         ? Player(PLAYER_NEUTRAL_AGGRESSIVE)
         : completingFaction.Player;
-
-      rescuer.RescueGroup(_rescueUnits);
     }
 
     /// <inheritdoc />
     protected override void OnComplete(Faction completingFaction)
     {
-      completingFaction.Player?
-        .RescueGroup(_rescueUnits);
+      SetPlayerTechResearched(completingFaction.Player, ResearchId, 1);
```

Comment:
You shouldn't need this - the ResearchId is always granted on completion. You can delete `OnComplete` entirely.